compiler/rustc_const_eval/src/interpret/memory.rs

```diff
@@ -1222,6 +1222,59 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
 
         Ok(())
     }
+
+    pub fn mem_swap_nonoverlapping(
+        &mut self,
+        x_ptr: Pointer<Option<M::Provenance>>,
+        y_ptr: Pointer<Option<M::Provenance>>,
+        count: u64,
+        layout: ty::layout::TyAndLayout<'tcx>,
+    ) -> InterpResult<'tcx> {
+        let elem_size = layout.size;
+        let align = layout.align.abi;
+
+        if count > i64::MAX as u64 {
+            throw_ub_format!("`count` argument to `swap_nonoverlapping_many` is too large.");
+        }
+
+        let first_ptr_acc = self.get_ptr_access(x_ptr, elem_size * count, align)?;
```

Comment:
Please use consistent naming: the arguments are `x` and `y` but here it becomes `first` and `second` and later you talk about "left" and "right". Stick to a single naming scheme.